mlir/lib/Dialect/MIOpen/Transforms/BlockwiseGemmToThreadwise.cpp

```diff
@@ -378,22 +220,29 @@ struct BlockwiseGemmRewritePattern : public OpRewritePattern<BlockwiseGemmOp> {
           lab.create<MulIOp>(loc, iva, MPerThreadSubCConstantOp)};
     }
 
-    // Emit threadwise_copy.
-    auto threadwiseCopyAMatrixOp = lab.create<ThreadwiseCopyOp>(
-        loc, op.matrixA(), threadAAllocOp,
-        b.getArrayAttr({transformsA, emptyArr}), noPadding, noOobDims,
-        noGlobals, matrixAThreadwiseCopySourceCoords,
-        matrixAThreadwiseCopyDestCoords);
-    affixThreadwiseCopyAttributes(threadwiseCopyAMatrixOp, op, b,
-                                  /*isMatrixA=*/true);
+    auto copyALoop = lab.create<TransformingForOp>(
+        loc,
+        ArrayRef<ValueRange>{matrixAThreadwiseCopySourceCoords,
+                             matrixAThreadwiseCopyDestCoords},
+        ArrayRef<Attribute>{transformsA, emptyArr},
+        ArrayRef<int64_t>{1, KPerThread, MPerThreadSubC},
+        /*forceUnroll=*/true, /*indexDiffs=*/false);
+    OpBuilder copyABuilder =
+        OpBuilder::atBlockTerminator(copyALoop.getBody(), lab.getListener());
+    Value aCopy = copyABuilder.create<memref::LoadOp>(
+        loc, matrixA, copyALoop.getLowerCoords(0));
```

Comment:
I feel `0` / `1` passed to `getLowerCoords` ought to be in an enumeration.
It seems in this PR we generally follow an idiom that `0` means the lower coordinates for the source to be read, and `1` means the lower coordinates for the destination to be stored.

Reply:
I'm not sure this is the right context for an enum - these are, in effect, indices into the arrays passed to the `TransformingForOp` builder.

Reply:
Yeah, I did deliberately choose that particular idiom, though fundamentally, nothing would go wrong if we went for something like "buffers first, then indexing into the vectors" as the ordering instead.

Reply:
Yes, and the use of `0` and `1` are almost universally visible in every call site after a `transforming_for` is emitted. Without a proper documentation of `getLowerCoords` it's hard to grasp what those values are for.

Would it be possible to do a key/value pair mapping to make the idea of `domains` more visible?

Reply:
I'll argue it's very visible in the printed IR - each iteration domain is its own `(%argX, %argY, %argZ) = [#transform_mapA, #transform_mapB, ...](%start1, %start2)` section in the printed loop. And this has some symmetry with `getBody([unsigned int block])` on `Operation`.

Though I do agree it isn't immediately obvious from the code which one's zero and which one's one - you have to look back at the constructor for the loop to get that.

... then again, you have to look back at the loop constructor to know which iter_arg is which.

Reply:
Good idea  - does it seem to work?